File: python/src/acedatacloud/resources/adsl.py
"""ADSL Direct Connection Proxy resources."""

from __future__ import annotations

from typing import Any, Literal


class Adsl:
    """Synchronous ADSL proxy client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def extract(
        self,
        *,
        isp: Literal["电信", "联通", "移动"] | str | None = None,
        city: str | None = None,
        type: Literal["http", "socks"] | None = None,
        dedup: bool | None = None,
        number: int | None = None,
        duration: Literal[60, 180, 300] | None = None,
        province: str | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {**kwargs}
        if isp is not None:
            body["isp"] = isp
        if city is not None:
            body["city"] = city
        if type is not None:
            body["type"] = type
        if dedup is not None:
            body["dedup"] = dedup
        if number is not None:
            body["number"] = number
        if duration is not None:
            body["duration"] = duration
        if province is not None:
            body["province"] = province
        return self._transport.request("POST", "/adsl/extract", json=body)

    def whitelist(
        self,
        *,
        ip: str | None = None,
        action: Literal["add", "delete"] | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {**kwargs}
        if ip is not None:
            body["ip"] = ip
        if action is not None:
            body["action"] = action
        return self._transport.request("POST", "/adsl/whitelist", json=body)


class AsyncAdsl:
    """Async ADSL proxy client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def extract(
        self,
        *,
        isp: Literal["电信", "联通", "移动"] | str | None = None,
        city: str | None = None,
        type: Literal["http", "socks"] | None = None,
        dedup: bool | None = None,
        number: int | None = None,
        duration: Literal[60, 180, 300] | None = None,
        province: str | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {**kwargs}
        if isp is not None:
            body["isp"] = isp
        if city is not None:
            body["city"] = city
        if type is not None:
            body["type"] = type
        if dedup is not None:
            body["dedup"] = dedup
        if number is not None:
            body["number"] = number
        if duration is not None:
            body["duration"] = duration
        if province is not None:
            body["province"] = province
        return await self._transport.request("POST", "/adsl/extract", json=body)

    async def whitelist(
        self,
        *,
        ip: str | None = None,
        action: Literal["add", "delete"] | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {**kwargs}
        if ip is not None:
            body["ip"] = ip
        if action is not None:
            body["action"] = action
        return await self._transport.request("POST", "/adsl/whitelist", json=body)

File: typescript/src/resources/adsl.ts
/** ADSL Direct Connection Proxy resources. */

import { Transport } from '../runtime/transport';

export class Adsl {
  constructor(private transport: Transport) {}

  async extract(opts: {
    isp?: '电信' | '联通' | '移动' | (string & {});
    city?: string;
    type?: 'http' | 'socks';
    dedup?: boolean;
    number?: number;
    duration?: 60 | 180 | 300;
    province?: string;
    [key: string]: unknown;
  } = {}): Promise<Record<string, unknown>> {
    const { isp, city, type, dedup, number, duration, province, ...rest } = opts;
    const body: Record<string, unknown> = { ...rest };
    if (isp !== undefined) body.isp = isp;
    if (city !== undefined) body.city = city;
    if (type !== undefined) body.type = type;
    if (dedup !== undefined) body.dedup = dedup;
    if (number !== undefined) body.number = number;
    if (duration !== undefined) body.duration = duration;
    if (province !== undefined) body.province = province;
    return this.transport.request('POST', '/adsl/extract', { json: body });
  }

  async whitelist(opts: {
    ip?: string;
    action?: 'add' | 'delete';
    [key: string]: unknown;
  } = {}): Promise<Record<string, unknown>> {
    const { ip, action, ...rest } = opts;
    const body: Record<string, unknown> = { ...rest };
    if (ip !== undefined) body.ip = ip;
    if (action !== undefined) body.action = action;
    return this.transport.request('POST', '/adsl/whitelist', { json: body });
  }
}
